### DISCUSSION TITLE
How to connect to microsoft power bi

### DISCUSSION THREAD

**Discussion by Siddharth-Changede-Hts** [supabase/supabase#13885] [Questions]
Could you help me  to connect supabase to power bi

**Comment by smedina1304** [comment]
Hello,

In *settings* you can find **Postgres Database** connection data.

<img width="588" alt="Screenshot_1" src="https://github.com/supabase/supabase/assets/46385135/373ad002-a85d-482c-a753-300fe891e84c">

In Power BI, you can use this link to help configure the Postgres connection.
https://blog.skyvia.com/postgresql-and-powerbi-connection/ 

Image from link above:
![power-bi-postgres-default-1](https://github.com/supabase/supabase/assets/46385135/c681c40f-5965-4a8f-8e34-31bf08b0e564)

Hope this helps!!!

**Reply by Vinith-AI** [reply]
I connected through the SSL Connection by importing certificate . Tried Connecting my supabase database and it throws this error .

<img width="576" height="255" alt="image" src="https://github.com/user-attachments/assets/2ddda314-f94f-4187-b1cc-cacc40a0cb56" />


Can Any one Help ?

**Comment by macro-id** [comment]
'The remote certificate is invalid according to the validation procedure

**Reply by FamosaNegra** [reply]
and you would need to update the postgres odbc driver to the older version


https://www.postgresql.org/ftp/odbc/versions/msi/

**Reply by RobinFierens** [reply]
What version of the odbc driver should we use?

**Reply by jotradovec** [reply]
@FamosaNegra  That link is not working anymore :{

**Comment by LordCortes** [comment]
Tube success in connecting? I have an invalid certificate error, I have tried several things and I cannot connect

**Comment by ariba932** [comment]
I had similar issue and resolved it by installing the certificate on my local computer -via mmc on trusted certificate authorities store.  (If certificate is missing - add via file --> add/remove snap-in.  Then ensure local or my computer option is selected. After adding certificate. Click on certificate  - > then select root Trusted Root Certificate Authorities. ---> at right side of the trusted root certification action - all tasks -- import (import certificate from supabase). 

Better still follow - this URL link (although issue was on general postgres issues)
The solution was found here: [https://social.technet.microsoft.com/Forums/windowsserver/en-US/85f2cdbf-f316-40a2-968b-ab823bffc947...](https://social.technet.microsoft.com/Forums/windowsserver/en-US/85f2cdbf-f316-40a2-968b-ab823bffc947/security-problem-with-vpn-sstp-error-0x0800b0109?forum=winserversecurity)

**Reply by jotradovec** [reply]
@ariba932  Where do you get the certificate in Supabase?

**Reply by jotradovec** [reply]
Oh, I was blind. It is here: Supabase > Your database > Configuration > Database > SSL

**Comment by Dark-Akkula** [comment]
Great responses above but I had chat GPT tell me step by step on this and it worked.  The only thing that was troublesome for me was figuring out that the certificate is generated within Supabase in the same section where you get the database connection.  Here is the step by step instructions from chat gpt but note my comments about generating the certificate in Supabase.

In a modern Windows environment, you can follow these steps to install a root certificate into the computer store:

Open MMC (Microsoft Management Console):

Press Win + R to open the Run dialog.
Type mmc and press Enter.
Add the Certificates Snap-in:

In the MMC console, go to File > Add/Remove Snap-in.
Select "Certificates" from the list of available snap-ins and click "Add".
Select Computer Account:

In the Certificates snap-in dialog, select "Computer account" and click "Next".
Choose "Local computer" and click "Finish".
Click "OK" to close the Add or Remove Snap-ins dialog.
Navigate to the Trusted Root Certification Authorities:

Expand the "Certificates" node in the left pane.
Navigate to Certificates > Trusted Root Certification Authorities in the console tree.
Import the Root CA Certificate:

Right-click on the "Trusted Root Certification Authorities" node.
Select All Tasks > Import to start the Certificate Import Wizard.
Follow the Certificate Import Wizard:

Click "Next" to begin the wizard.
Browse to the location of the root CA certificate file (.cer) and select it.
Click "Next" to proceed.
Choose "Place all certificates in the following store" and ensure "Trusted Root Certification Authorities" is selected.
Click "Next" to continue.
Review the summary and click "Finish" to complete the import process.
Confirmation:

You should receive a confirmation message indicating that the import was successful.
Click "OK" to close the confirmation dialog.
After completing these steps, the root certificate should be installed into the Trusted Root Certification Authorities store for the local computer. This should enable the computer to trust certificates issued by the root CA when establishing secure connections, including VPN connections like SSTP.

**Reply by isaac-ros** [reply]
It worked, thanks a lot!

**Reply by jotradovec** [reply]
Note that the certificate can be downloaded in Supabase > Your database > Configuration > Database > SSL

**Reply by Vinith-AI** [reply]
I connected through the SSL Connection by importing certificate . Tried Connecting my supabase database and it throws this error .

<img width="576" height="255" alt="image" src="https://github.com/user-attachments/assets/2ddda314-f94f-4187-b1cc-cacc40a0cb56" />


Can Any one Help ?

**Reply by Ellba** [reply]
@Vinith-AI "server signature not received" is most likely the wrong database password. You can update db password (postgres role) in Supabase Studio here - https://supabase.com/dashboard/project/_/database/settings

**Comment by Weldawadyathink** [comment]
Are there any solutions that don't involve adding a new root certificate? I am hoping to use this on the hosted PowerBI service, and I am almost positive adding new certificates is not an option.

**Comment by hugomoran159** [comment]
What worked for my was disabling encrypt connection in the power bi credential settings. 

File -> Options and Settings -> Data Source Settings -> Edit Permissions -> Uncheck encrypt connections

**Reply by enzovillafuerte** [reply]
This worked for me too!

**Reply by automacaotehokas** [reply]
é essa a solução definitiva!!!! Funciona mesmo

**Reply by salomaoparkour** [reply]
Perfect 🙏 Thank you!!

**Comment by lenovo-ernest** [comment]
Downloaded the supabase certificate
Installed it. During installation I selected "Local Computer". Manually specified "trusted certificates" or something like that. After that everything worked
![image](https://github.com/user-attachments/assets/740e46b4-0a1b-4367-86dc-b035adf3c198)

**Comment by hatedabamboo** [comment]
I was able to configure the connection from Power BI hosted service to Supabase database and keep the connection encrypted.

To achieve this you will need an [on-premise gateway](https://www.microsoft.com/en-us/power-platform/products/power-bi/gateway). It can be installed locally or on AWS ec2 Windows instance and then it should be connected to your Microsoft BI account.
After that you should navigate to database settings in Supabase and download the certificate (as mentioned in the [comment](https://github.com/orgs/supabase/discussions/13885#discussioncomment-10327768) above).
This certificate should be installed to the machine where the gateway is running (either on local Windows or on EC2, instructions have already been provided above).
After that in your BI web interface try to configure a new database connection, but instead of „cloud” option select „on premise gateway” and the configuration parameters of your database.

This solution worked for me and hopefully will prove useful to you.

TLDR:
- Set up BI on-premise gateway
- Install Supabase certificate to the host with the on-premise gateway
- Configure in BI web new connection via on-premise gateway

**Reply by Jamil1016** [reply]
I try this but the result is still this
"Unable to create connection for the following reason: Unable to connect to the data source. Either the data source is inaccessible, a connection timeout occurred, or the data source credentials are invalid. Please verify the data source configuration and contact a data source administrator to troubleshoot this issue.

Details:

Personal: An error happened while reading data from the provider: 'The remote certificate is invalid according to the validation procedure.'

Please have this information handy if you choose to create a support ticket.

Session Id: d36b3097-a22a-43ce-8563-81943ca34f49
RequestId: 03adbc93-6eb4-4642-804e-3e3986793b54
Cluster URI: https://api.powerbi.com
Status code: 400
Time: Sat Jul 19 2025 23:40:39 GMT+0800 (Singapore Standard Time)"

**Comment by maxprime1** [comment]
I was able to connect to my database in supabase. I'm able to view all the schemas and tables in the navigator but it keeps evaluating when I try to load the data without any end in sight. Is there any setting in supabase side that I need to configure to load the tables?

[a comment by an automated account is not shown]

**Reply by lucasmarcuzo** [reply]
here it works, thanks!

**Comment by automacaotehokas** [comment]
Thank you for replying to me almost a month later. I figured it out on my own.



[cid:042794c5-c687-4ce6-8eca-15a64778494b]

[linkedin]<https://www.linkedin.com/in/tehokas-solu%C3%A7%C3%B5es-2b5b98254/>

[instagram]<https://tehokas.solucoes/>

Eric Martins

Soluções Para Clientes

Gestão de Processos e Projetos | Tehokas Soluções

[mobilePhone]

38 98827-6452<tel:38%2098827-6452>

[website]

www.tehokas.com.br<https://www.tehokas.com.br/>





________________________________
De: joris-taes-ci ***@***.***>
Enviado: sexta-feira, 23 de maio de 2025 06:47
Para: supabase/supabase ***@***.***>
Cc: Eric Martins ***@***.***>; Comment ***@***.***>
Assunto: Re: [supabase/supabase] How to connect to microsoft power bi (Discussion #13885)


For those struggling:

db..supabase.co
postgres

postgres

confirm, you will get an error!
Go to data source settings, global, edit permission, turn off encryption.

**Comment by darylteo** [comment]
Sorry but this does not work (at least, for cloud hosted service).

There's the client side configuration which allows you to view and query the data. 

But the Refresh service does not respect the "Use Encrypted Connections" checkbox when unchecked. So when trying to refresh the data through the background refresh service, it still does not work. 

```
Underlying Error DataSource.Error: An error happened while reading data from the provider: 'The remote certificate is invalid according to the validation procedure.'. Microsoft.Data.Mashup.ErrorCode = 10434. Microsoft.Data.Mashup.ValueError.DataSourceKind = PostgreSQL. DataSourcePath = db.dcrldhbbnfunlpulavio.supabase.co;postgres. . The exception was raised by the IDbCommand interface. Table: analytics bluehippo_machines. Partition: analytics bluehippo_machines.
Activity ID c817eca4-3c6e-3d6d-a93a-dd4751e190c1
Time 2026-03-16T02:20:24.751263Z
```

I've also tried the Direct Connection, and Pooler. And even with the IPv4 Add-On. 

Unfortunately this means that PowerBI simply cannot support Supabase due to custom SSL certficate without the stupid gateway

Not blaming Supabase btw! ChatGPT/Claude brought me down a rabbit hole promising solutions so I feel compelled to leave a message here for the future.

**Reply by salomaoparkour** [reply]
I can confirm this behavior. And even after disabling encrypted connections, it still doesn't sync. I've tried it multiple times, with different projects and databases.
